hs-bindgen-runtime/src/HsBindgen/Runtime/CEnum/General.hs

```diff
@@ -0,0 +1,305 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module HsBindgen.Runtime.CEnum.General (
+    -- * Type Class
+    GeneralCEnum(..)
+    -- * Bounded API
+  , generalCEnumMinBound
+  , generalCEnumMaxBound
+    -- * Enum API
+  , generalCEnumSucc
```

Comment:
Could hide these, would provide for more implementation hiding, giving us the ability to change internals whilst keeping all functions still mutually consistent.